### ISSUE TITLE
[Bug]: Unnecessary warning generated for service-decl resources with non-public symbols

### ISSUE THREAD

**Issue by hasithaa** [ballerina-platform/ballerina-lang#38460]
### Description

The generated warning is "attempt to expose non-public symbol"

- For network interactions and data binding, the shape of the data is enough. The type name is not required.  The type name is an internal representation.
- This warning adds unnecessary complexity to the service developer.
- Tools like OpenAPI, DocGen should be able to handle this without breaking existing functionalities. 
  - For e.g.: For doc generation, if a type is not public, it can render the shape of the type without exposing the type name.

Remove this warning for service-decl for now. 

We should revisit, where should we generate this warning and it will be captured as part of the #30324 

### Steps to Reproduce

Source code used. 
```ballerina
import ballerina/http;
service / on new http:Listener(9090) {
    isolated resource function post hello (@http:Payload Foo foo) returns error?{
        
    }
}

type Foo record {|
 int x;
 string y;
|};
```

### Affected Version(s)

This is observed in `2202.2.0` `2202.2.2`, Not in Update 2201.1.1. 

### Related area

-> Compilation

**Comment by hasithaa** [comment]
@KavinduZoysa We need to fix this, where we validate "attempt to expose non-public symbol" warning. Skip this validation for service-decl.

**Comment by KavinduZoysa** [comment]
Fixed via #38476

**Comment by ThisaruGuruge** [comment]
Do this apply to the objects as well?

For example, in a Ballerina GraphQL service, we can return a service type. Currently, we see this warning all the places. Adding more to the complexity, when we mark a service type as a public type, then the resource functions are complaining about isolation since it does not infer the isolated property for public types.
